src/spandrel/architectures/CRAFT/__init__.py

```diff
@@ -0,0 +1,83 @@
+import re
+from ..__arch_helpers.state import get_scale_and_output_channels
+from ...__helpers.model_descriptor import (
+    ImageModelDescriptor,
+    SizeRequirements,
+    StateDict
+)
+
+from .arch.CRAFT import CRAFT
+
+def load(state_dict: StateDict) -> ImageModelDescriptor[CRAFT]:
+    # Required values
+    in_chans = state_dict['conv_first.weight'].shape[1]
+    embed_dim = state_dict['layers.0.conv.weight'].shape[1]
+
+    depths = []
+    num_heads = []
+
+    split_size_0 = 4
+    split_size_1 = 16
+
+    mlp_ratio = 2.
+
+    qkv_bias = True
+    qk_scale = None
+
+    # norm_layer=nn.LayerNorm
+
+    upscale = int(math.sqrt(state_dict['upsample.0.bias'].shape[0] / in_chans))
+    img_range = 1.
+    resi_connection = '1conv'
+
+    for key, tensor in state_dict.items():
+        depth_match = re.search('layers.(\d+).residual_group.srwa_blocks.(\d+).norm1.weight', key)
```

Comment:
You'll want to `r`-prefix this string so `\d` doesn't smell like an invalid escape:
```suggestion
        depth_match = re.search(r'layers.(\d+).residual_group.srwa_blocks.(\d+).norm1.weight', key)
```